packages/events/src/router/middleware/authorization/index.ts

```diff
@@ -254,3 +256,32 @@ export const requireScopeForWorkqueues: MiddlewareFunction<
   }
   return next()
 }
+
+/**
+ * Checks that the token has been exchanged for the specific `eventId` and `actionId` in the input.
+ *
+ * Registrars token can be exchanged in auth into a more specific token with `eventId` and `actionId`.
+ * This is useful when tokens need to be exposed outside of core of OpenCRVS, e.g. countryconfig or external systems.
+ */
+export const requireEventActionAuthorization: MiddlewareFunction<
+  TrpcContext,
+  OpenApiMeta,
+  TrpcContext,
+  TrpcContext,
+  ActionConfirmationResponseSchema
+> = async ({ next, ctx, input }) => {
+  const { eventId: grantedEventId, actionId: grantedTokenId } = getTokenPayload(
+    ctx.token
+  )
+
+  if (!grantedEventId && !grantedTokenId) {
+    // This was not an exchanged token (i.e. a user token), so we skip this check
+    return next()
```

Comment:
Just thinking out loud: do we need to allow normal user tokens to make requests to these accept/reject endpoints at all? 🤔 Or could we assume that it will always be done with the exchanged token?